src/System.Drawing.Common/tests/IconTests.cs

```diff
@@ -495,40 +492,96 @@ public void ToBitmap_BitmapIcon_Success(Icon icon)
         }
 
         [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
-        public void ToBitmap_PngIcon_Success()
+        public void ToBitmap_BitmapIconFromHandle_ReturnsExpected()
         {
-            Icon icon;
-            using (var stream = new MemoryStream())
+            // Handle rerring to icon without any colour.
+            using (var originalIcon = new Icon(Helpers.GetTestBitmapPath("48x48_one_entry_1bit.ico")))
+            using (Icon icon = Icon.FromHandle(originalIcon.Handle))
             {
-                // Create a PNG inside an ICO.
-                var bitmap = new Bitmap(10, 10);
-                stream.Write(new byte[] { 0, 0, 1, 0, 1, 0, (byte)bitmap.Width, (byte)bitmap.Height, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 22, 0, 0, 0 }, 0, 22);
+                ToBitmap_BitmapIcon_ReturnsExpected(icon);
+            }
+        }
 
-                // Writing actual data
-                bitmap.Save(stream, ImageFormat.Png);
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ToBitmap_PngIconSupportedInSwitches_Success()
+        {
+            RemoteInvoke(() =>
```

Comment:
Instead of always going out of proc and checking for isfullframework you could instead check GetSwitch and only go out of proc (and set to false) if it's true.
That's the "right" way to deal with appcontext switches  -- to check the switch, not the version (platform, in this case).